### ISSUE TITLE
ArrayIndexOutOfBoundsException in CardDataProvider

### ISSUE THREAD

**Issue by desperateCoder** [stefan-niedermann/nextcloud-deck#789]
original post: https://github.com/stefan-niedermann/nextcloud-deck/issues/782#issuecomment-756752358

```
App Version: 1.13.5
App Version Code: 1013005
Server App Version: 1.2.2
App Flavor: play

Files App Version Code: 30140190

---

OS Version: 4.4.23+(354(C432))
OS API Level: 26
Device: HWRNE
Manufacturer: HUAWEI
Model (and Product): RNE-L21 (RNE-L21)

---

java.lang.ArrayIndexOutOfBoundsException: length=22; index=22
	at java.util.ArrayList.add(ArrayList.java:468)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider$1.onResponse(CardDataProvider.java:53)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider$1.onResponse(CardDataProvider.java:50)
	at it.niedermann.nextcloud.deck.api.RequestHelper$ResponseConsumer.accept(RequestHelper.java:62)
	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher$PublisherSubscriber.onNext(ObservableFromPublisher.java:56)
	at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:111)
	at com.nextcloud.android.sso.api.-$$Lambda$NextcloudAPI$8WS9mapJbOYsPgbhBi25vBNYZqI.subscribe(Unknown Source:6)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:578)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:66)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:57)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:301)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
	at java.lang.Thread.run(Thread.java:784)

```

**Comment by desperateCoder** [comment]
@MediaPlusPL I think I found the problem. We were modifying a list from multiple threads at the same time. The fix will be shipped with the next release.

**Comment by stefan-niedermann** [comment]
@MediaPlusPL Your second issue has been fixed in `1.13.6`. I just pushed this version to the Play Store beta channel. Depending on Google it should be available for you in the next few hours.
